addons/udes_stock/models/stock_picking.py

```diff
@@ -1879,6 +1879,51 @@ def _get_suggested_location_by_products(self, move_line_ids, products=None):
 
         return suggested_locations
 
+    def _get_suggested_location_by_product_lot(self, move_line_ids, products=None):
+        """Get suggested locations by matching both product and lot (batch) number.
+
+        If used on a product that is not tracked by lot then locations will be suggested
+        based on products.
+        """
+        Quant = self.env["stock.quant"]
+
+        if not move_line_ids:
+            raise ValidationError(_("Cannot determine the suggested location: missing move lines"))
+
+        self._check_picking_move_lines_suggest_location(move_line_ids)
+
+        if products is None:
+            products = move_line_ids.mapped("product_id")
+
+        if not products:
+            raise ValidationError(_("Products missing to suggest location for."))
+        elif len(products) > 1:
+            raise ValidationError(_("Cannot drop different products by lot (batch) number."))
+
+        lots = move_line_ids.filtered(lambda ml: ml.product_id in products).mapped("lot_id")
+        if len(lots) > 1:
+            raise ValidationError(_("Cannot drop products with different lot (batch) numbers in a "
+                                    "single package/pallet when dropping by product and batch."))
+
+        quant_domain = [
+            ("product_id", "in", products.ids),
+            ("location_id", "child_of", self.location_dest_id.ids),
+            ("location_id.u_blocked", "=", False),
+            ("location_id.barcode", "!=", False),
+        ]
+        if lots:
+            quant_domain.append(("lot_id", "in", lots.ids))
+        else:
```

Comment:
Is this really needed ?

Reply:
I guess technically not. The only case it may be useful is as a failsafe incase a product has changed from not tracked to lot-tracked. This way quants without lots are lumped together rather than being mixed with the those with lots

Reply:
Removed this as it's probably too much of an edge case